### PR TITLE
Command delete event

### PR DESCRIPTION
**Pull request by taeewonnn**

### CODE REVIEW

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -21,6 +21,7 @@ public class Messages {
                 "Multiple values specified for the following single-valued field(s): ";
     public static final String MESSAGE_INVALID_EVENT_DISPLAYED_INDEX = "The event index provided is invalid";
     public static final String MESSAGE_SELECT_EVENT = "No event is selected!";
+    public static final String MESSAGE_INVALID_DELETE_EVENT = "No event can be deleted when it is selected";
```
Comment (Apzure) [inline src/main/java/seedu/address/logic/Messages.java:24]: Put this inside DeleteCommand.java, as it is specific to DeleteCommand.
Change it to "The selected Event cannot be deleted"

File: src/main/java/seedu/address/logic/commands/DeleteEventCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Event> lastShownList = model.getFilteredEventList();
+
+        if (model.isAnEventSelected()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_DELETE_EVENT);
```
Comment (Apzure) [inline src/main/java/seedu/address/logic/commands/DeleteEventCommand.java:40]: The command should only not work when delev is trying to delete the event that it is currently selected, in other words
it is okay to delete events that are not selected.

So you should make a method in modelmanager.java to check whether the event that it is trying to delete is the event that is currently selected. Sorry if my phrasing was vague before.

**Review by Apzure — CHANGES_REQUESTED** [review]
Overall looks good, just 2 changes. I'll leave the testing to Max.

File: src/main/java/seedu/address/model/Model.java [hunk 3]
```diff
@@ -100,6 +100,8 @@ public interface Model {
 
     boolean isAnEventSelected();
 
+    boolean isEqualSelectedEvent(Event event);
```
Comment (ChocoRagdoll) [inline src/main/java/seedu/address/model/Model.java:103]: I feel like isSameSelectedEvent also works but well this surely works okay

File: src/test/java/seedu/address/logic/commands/DeleteEventCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Test
+    public void execute_selectModel_throwsCommandException() {
+        SelectCommand selectCommand = new SelectCommand(Index.fromZeroBased(1));
+
+        assertThrows(CommandException.class, () -> new DeleteEventCommand(Index.fromZeroBased(1)).execute(model));
+    }
+
+    @Test
+    public void execute_deleteEventSuccessful() throws CommandException {
```
Comment (ChocoRagdoll) [inline src/test/java/seedu/address/logic/commands/DeleteEventCommandTest.java:47]: shld be execute_deleteEvent_success() to ensure consistency with the existing code

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 5]
```diff
@@ -54,6 +56,13 @@ public void parseCommand_delete() throws Exception {
         assertEquals(new DeleteCommand(INDEX_FIRST_PERSON), command);
     }
 
+    @Test
+    public void parseCommand_deleteEvent() throws Exception {
```
Comment (ChocoRagdoll) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:60]: _success

**Review by ChocoRagdoll — APPROVED** [review]
lgtm, just some cosmetic changes needed, after that can merge alr. Does max need to check the tests?

File: src/main/java/seedu/address/logic/commands/DeleteEventCommand.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import seedu.address.model.event.Event;
+
+/**
+ * Deletes an event identified using it's displayed index from the address book.
+ */
+public class DeleteEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "delev";
+
+    public static final String MESSAGE_INVALID_DELETE_EVENT = "The selected Event cannot be deleted";
```
Comment (Apzure) [inline src/main/java/seedu/address/logic/commands/DeleteEventCommand.java:21]: After further thought, change this to "Cannot delete the event that is currently selected", it is more clear that way

File: src/main/java/seedu/address/model/Model.java (hunk 3 above)
Comment (Apzure) [reply src/main/java/seedu/address/model/Model.java:103]: Is there any reason why this name differs from the method name in EventBook? If not, make them have the same name for readability

**Review by Apzure — CHANGES_REQUESTED** [review]

**Review by Apzure — APPROVED** [review]
LGTM

**Review by myang2020 — APPROVED** [review]
lgtm

**Pull request merged**